src/profile-logic/marker-schema.js

```diff
@@ -394,8 +404,118 @@ export function formatFromMarkerSchema(
+        `A marker schema of type "${markerType}" had an unknown format ${JSON.stringify(
+          format
+        )}`
       );
       return value;
   }
 }
+
+export function isDOMRenderingFormat(format: MarkerFormatType): boolean {
+  return format === 'url' || typeof format === 'object' || format === 'list';
+}
+
+function _formatDOMFromMarkerSchema(
+  markerType: string,
+  format: MarkerFormatType,
+  value: any,
+  // add elements that allow interaction (like clickable urls)
+  supportsInteraction: boolean
+) {
+  if (!isDOMRenderingFormat(format)) {
+    return <>{formatFromMarkerSchema(markerType, format, value)}</>;
+  }
+  if (typeof format === 'object') {
+    switch (format.type) {
+      case 'table': {
+        const { columns } = format;
+        if (!(value instanceof Array)) {
+          throw new Error('Expected an array for table type');
+        }
+        const hasHeader = columns.some((column) => column.label);
+        return (
+          <table>
+            {hasHeader ? (
+              <tr>
+                {columns.map((col, i) => (
+                  <th key={i}>{col.label || ''}</th>
+                ))}
+              </tr>
+            ) : null}
+            {value.map((row, i) => {
+              if (!(row instanceof Array)) {
+                throw new Error('Expected an array for table row');
+              }
+
+              if (row.length !== columns.length) {
+                throw new Error(
+                  "Row length doesn't match column count (row: " +
+                    row.length +
+                    ', cols: ' +
+                    columns.length +
+                    ')'
+                );
+              }
+              return (
+                <tr key={i}>
+                  {row.map((cell, i) => {
+                    return (
+                      <td key={i}>
+                        {_formatDOMFromMarkerSchema(
+                          markerType,
+                          columns[i].type || 'string',
+                          cell,
+                          supportsInteraction
+                        )}
+                      </td>
+                    );
+                  })}
+                </tr>
+              );
+            })}
+          </table>
+        );
+      }
+      default:
+        throw new Error(`Unknown format type ${JSON.stringify(format)}`);
+    }
+  }
+  switch (format) {
+    case 'list':
+      if (!(value instanceof Array)) {
+        throw new Error('Expected an array for list format');
+      }
+      return value.map((entry, i) => (
+        <li key={i}>
+          {formatFromMarkerSchema(markerType, 'string', value[i])}
+        </li>
+      ));
+    case 'url':
+      if (supportsInteraction) {
+        return (
+          <a href={value}>
+            {formatFromMarkerSchema(markerType, 'string', value)}
+          </a>
+        );
+      }
+      return <>value</>;
```

Comment:
(if we keep `supportsInteraction`) You should probably call and return the result of `formatFromMarkerSchema`. We don't need the fragment either, the fragment is useful when you have to wrap several nodes, but here we have only one.